### ISSUE TITLE
USB C Ethernet not working; slow wifi; wifi disconnects after sleep; occasional wake ups during sleep

### ISSUE THREAD

**Issue by baglana** [daliansky/XiaoMi-Pro-Hackintosh#755]
**State the model**
XiaoMi-Pro MX150 i5-8G-256G

**State the version of the EFI**
v1.8.1

**Describe the bug**

1. USB type c Ethernet adapter but it doesn't work

USB 10/100 LAN is listed as Disconnected in network settings:
![image](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/102351104/7f5babf3-e842-45ad-af0d-8cd9bacb64c9)

Details screen is like this:
![image](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/102351104/83ff12d8-4fe7-4d65-b6ee-3a094c20fc9f)

Tried installing the driver referenced here https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/256#issuecomment-540884088
Downloaded the RTUNICv1.0.22_Update4.zip file from https://www.realtek.com/en/component/zoo/category/network-interface-controllers-10-100-1000m-gigabit-ethernet-usb-3-0-software
when the page was still available.

Tried multiple times installing the driver and restarting laptop but it didn't solve the issue. Although the installer says that the driver was installed successfully, it seems that it is not.

I also tried running the following commands mentioned here https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/275#issuecomment-540709812
```
sudo kextunload /Library/Extensions/AppleRTL815XEthernet109.kext
sudo kextunload /Library/Extensions/AppleRTL815XComposite109.kext
sudo kextload /Library/Extensions/AppleRTL815XEthernet109.kext
sudo kextload /Library/Extensions/AppleRTL815XComposite109.kext
```

But  there is no /Library/Extensions/AppleRTL815XEthernet109.kext file on my laptop. Instead there is a /Library/Extensions/AppleRTL815XEthernet110.kext file. So I tried the previous set of commands with updated filenames as well but still get errors:

![image](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/102351104/5ff61f41-4f1a-49d5-8357-237762852a71)

I suspect that the issue might be with my version of the USB c Ethernet adapter being 8152 while in the mentioned issues people had 8153 (and the driver I tried is for 8152B/8153/8156 but not for 8152):

![image](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/102351104/369675b0-abf6-4dc3-b01f-71861c94c392)
![telegram-cloud-document-2-5309826552890936894](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/assets/102351104/b609354b-7b30-4e8e-b42e-89d7e33c1e0b)

Do you know what might be the problem? Is there anyone with RTL8152 who has working ethernet? If I buy an RTL8153 adapter will it work?

2. Does the 

> Intel Wi-Fi works with low performance

in the repo readme mean that built in wifi works slowly? Mine works around times slower than wifi on another laptop in the worst case and around 50% slower in the best case. Is it ok? The only fix would be to buy a USB wifi dongle?

3. After sleep the wifi connection occasionally gets disconnected and I have to reconnect manually. Is it a known issue and is there a fix for that?

Maybe I should update the EFI from v1.8.1 to v1.8.2 to solve these problems? Also, does upgrading EFI to a newer version mean that macOS will update? If something will go wrong after I update the EFI folder, is there an easy way to rollback and not lose existing data?

Sorry if my questions are stupid, I am not tech savvy and know almost nothing about hackintosh.

**System log**
- If a windows "Your computer was restarted because of a problem" appears, click `Report` and attach the log to help find the problem.

**Screenshots**
- If applicable, add screenshots to help explain your problem.

**Attach your entire EFI folder**
- Please attach your entire EFI folder to help solve your problem.
- Erasing serial numbers in config.plist is required!
**Attach system.log**
- If applicable, please run `log show --predicate "processID == 0" --last boot --debug | grep "ERR"` in `Terminal.app` to help locate the problem.

**Comment by stevezhengshiqi** [comment]
1. Many third-party ethernet drivers are removed in Sonoma, and it might affect your Realtek model. I would recommend purchase a USB-C Ethernet Adapter that can work out of box, like the Xiaomi brand (reported working a few years ago, no guarantee)
2. Yes, I would recommend updating to EFI v1.8.2 since the Intel driver improves itself consistently according to [itlwm's Changelog](https://github.com/OpenIntelWireless/itlwm/releases). Also, v1.8.2 fixed an Intel WiFi build logic issue.
3. I am not that sure about the Intel WiFi's behavior as different users have different router setup. Keep an eye on Intel WiFi's issue page could help: https://github.com/OpenIntelWireless/itlwm/issues
4. The regular awake is also happening on real Macs. There are multiple process like rtc awake, `com.apple.alarm*` would break the sleep.

**Comment by baglana** [comment]
Thanks for your answers.

1. Are the USB-C Ethernet Adapters that can work out of the box those that work with real Macs? So the only important thing for the adapter to work is that it has to be compatible with macOS and it is not laptop hardware dependent? I searched for Xiaomi ethernet adapters and could only find USB A version adapters in stock. It seems like Xiaomi has discontinued their production. Does it matter if it is a type A or type C adapter?
2. Thanks, I updated the EFI to v1.8.2 and the WIFI speed has doubled. Now it is on par with my other laptop WIFI speed.

**Comment by stevezhengshiqi** [comment]
> 1. Are the USB-C Ethernet Adapters that can work out of the box those that work with real Macs?

Yes, for third-party peripheral driver, our hack behaves exactly the same as real Macs. The compatibility is determined by apple driver. If you are certain that the adapter work out of box for real Macs, it will also work for Xiaomi laptop. The only exception is Thunderbolt devices, which is not a concern for Xiaomi laptop anyway.
